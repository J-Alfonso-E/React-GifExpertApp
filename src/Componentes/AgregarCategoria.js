import React, { useState } from "react";
import PropTypes from 'prop-types';

export const AgregarCategoria = ({EnviarCategoria}) => { //Se recibe la funcion desde GifExpertApp

    const [ValorEntrada, SetValor] = useState('');

    const CambioValor = (e) => {
        SetValor(e.target.value);
        //console.log('Evento Cambio de Valor');
    }

    const EnvioCategoria = (e) => {
        e.preventDefault();
        if(ValorEntrada.trim().length > 2){
            
            EnviarCategoria(cats => [ValorEntrada, ...cats,]); //Cuando se recibe una funcion de Hook el Estado puede ser accedido de esta manera y agregar la nueva categoria
            SetValor('');
        }

        
    }

    return (
        <form onSubmit={EnvioCategoria}>
            <h2>Agregar Categoria</h2>
            <p> {ValorEntrada} </p> {/* Valor de Prueba */}
                <input
                type='text'
                value = {ValorEntrada}
                onChange={CambioValor}
                >
                </input>
        </form>
    )
};

AgregarCategoria.propTypes = {
    EnviarCategoria : PropTypes.func.isRequired
}