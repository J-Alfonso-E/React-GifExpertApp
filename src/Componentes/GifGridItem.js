import React from "react";
import PropTypes from 'prop-types';


export const GifGridItem = ({title, url}) => {
    console.log({title, url});

    return (
        <div className='card animate__fadeInDown'>
            <img className='card animate__bounceIn' src={url} alt={title}></img>
            <p className='card'>{title}</p>
        </div>
    )
}

GifGridItem.propTypes = {
    title: PropTypes.string.isRequired,
    url: PropTypes.string.isRequired
}