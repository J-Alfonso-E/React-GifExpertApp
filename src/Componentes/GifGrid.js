import React from "react";
import { useFetchGifs } from "../hooks/useFetchGifs";
//import { ObtGif } from "../Complementos/GifFetch";
import { GifGridItem } from "./GifGridItem";

import PropTypes from 'prop-types';

export const GifGrid = ( {categoria}) => {

    const {data, loading} = useFetchGifs( categoria);

    return (
        <>
        <h3>{categoria}</h3>

        {loading && 'Cargando' } {/* Operador Ternario donde si es true hace lo primero en caso contrario no hace nada */}
        <div className='card-grid animate__bounceIn'>
            
            
                {data.map( img => (
                    <GifGridItem key={img.id} {...img} />
                ))
                }
        </div>
        </>
    )
}

GifGrid.propTypes = {
    categoria: PropTypes.string.isRequired
}