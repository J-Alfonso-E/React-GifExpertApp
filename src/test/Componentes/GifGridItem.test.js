
import {shallow} from 'enzyme';
import { GifGridItem } from '../../Componentes/GifGridItem';



describe('Mostrando las Imagenes', () => {

    const imagen = 'https://media4.giphy.com/media/b4ab9TBk9Ornvrt9W8/giphy-downsized-medium.gif?cid=4e4c6c79eazgluaod3tfwrerbs3v77bfd6uivr8mkwsl357h&rid=giphy-downsized-medium.gif&ct=g';

    const titulo = 'Master Chief Halo GIF by Xbox';

    const wrapper = shallow(<GifGridItem title={titulo} url={imagen}/>);

    test('Prueba de imagenes', () =>{
        expect(wrapper).toMatchSnapshot();
    });

    test('Titulo de la imagen', () =>{
        const p = wrapper.find('p');
        expect(p.text().trim()).toBe(titulo);
    });

    test('Verificar que este la imagen con titulo', () =>{
        const img = wrapper.find('img');
        //console.log(img.prop().src); // Una forma de ver la propiedad deseada 
        expect(img.prop('src')).toBe(imagen); //Se puede acceder a las propiedades de mendiante las props
        expect(img.prop('alt')).toBe(titulo);
    });

    test('Verificar que el div tenga la clase animate__fadeInDown', () =>{
        const div = wrapper.find('div');
        const clase = div.prop('className');
        //console.log(img.prop().src); // Una forma de ver la propiedad deseada 
        //console.log(clase);
        
        expect(clase.includes('animate__fadeInDown')).toBe(true);
    });


});

