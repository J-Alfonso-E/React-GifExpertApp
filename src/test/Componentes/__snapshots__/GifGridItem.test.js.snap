// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Mostrando las Imagenes Prueba de imagenes 1`] = `
<div
  className="card animate__fadeInDown"
>
  <img
    alt="Master Chief Halo GIF by Xbox"
    className="card animate__bounceIn"
    src="https://media4.giphy.com/media/b4ab9TBk9Ornvrt9W8/giphy-downsized-medium.gif?cid=4e4c6c79eazgluaod3tfwrerbs3v77bfd6uivr8mkwsl357h&rid=giphy-downsized-medium.gif&ct=g"
  />
  <p
    className="card"
  >
    Master Chief Halo GIF by Xbox
  </p>
</div>
`;
